src/realm/db.cpp

```diff
@@ -1518,6 +1518,17 @@ void DB::close_internal(std::unique_lock<InterprocessMutex> lock, bool allow_ope
     }
 }
 
+bool DB::waiting_for_write_lock() const
+{
+    SharedInfo* info = m_file_map.get_addr();
+
+    uint32_t next_ticket = info->next_ticket.load(std::memory_order_relaxed);
+    uint32_t next_served = info->next_served.load(std::memory_order_relaxed);
```

Comment:
Yes, at least the way we use it.

But I can think of a scenario where a thread starts writing (i.e, acquires the lock) and checks if other threads are waiting for the lock while at the same time another thread closes the transaction causing `next_served` to be incremented. This could then cause some wrong results. Any thoughts @finnschiermer ?

Reply:
For future reference: the above scenario cannot happen. The only thread which can close the transaction is the one holding the lock.